sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -170,7 +170,15 @@ class OrcFileFormat
     (file: PartitionedFile) => {
       val conf = broadcastedConf.value.value
 
-      val filePath = new Path(new URI(file.filePath))
+      var path: Option[Path] = None
+      import scala.util.Try
```

Comment:
I would prefer to avoid relying on try-catch whenever possible.

Also, a space in a URI doesn't look correct. HDFS itself has a complicated behaviour arond path and URIs so let;s don't follow the behaviour unless it's known as a legitimate behaviour (is it?).`file.filePath` should be a URI so the behaviour `new Path(new URI(file.filePath))` seems correct.